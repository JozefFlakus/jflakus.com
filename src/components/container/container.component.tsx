import styled, { ColorBg } from '../../styles/theme.styles';

// Container
// -----------------

interface ContainerProps {
  bgColor?: ColorBg;
  spaced?: boolean;
}

export const Container = styled<ContainerProps, 'div'>('div')(
  ({ theme, bgColor, spaced }) => ({
    padding: spaced ? '100px 0' : 0,
    backgroundColor: !!(theme.colorBg) && theme.colorBg[bgColor],
  })
);

// Container-Main
// -----------------

export const ContainerMain = styled<{}, 'div'>('div')(
  ({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative',
    minHeight: '100vh',
    padding: `0 ${theme.dimensions.module * 2}px`,

    '&::after': {
      content: `''`,
      position: 'absolute',
      top: 0,
      bottom: 0,
      right: '50%',
      borderRightWidth: 1,
      borderRightStyle: 'solid',
      borderRightColor: theme.util.lineColor,
      zIndex: -1,
    },
  })
);
