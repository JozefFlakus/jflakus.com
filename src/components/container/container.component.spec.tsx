import * as React from 'react';
import * as emotion from 'emotion';
import { createSerializer } from 'jest-emotion';
import * as renderer from 'react-test-renderer';
import { Container } from './container.component';
import { testWithTheme } from '../../utils/test.util';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('<Container>', () => {
  it('renders correctly', () => {
    const component = testWithTheme(<Container>Test</Container>);
    const result = renderer.create(component).toJSON();
    expect(result).toMatchSnapshot();
  });
});
