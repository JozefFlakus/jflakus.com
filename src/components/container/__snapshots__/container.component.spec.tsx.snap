// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container> renders correctly 1`] = `
.emotion-0 {
  padding: 0;
}

<div
  className="emotion-0"
>
  Test
</div>
`;
